src/persist/src/azure.rs

```diff
@@ -0,0 +1,364 @@
+        // and need to be refreshed. This can be done through `service_client.update_credentials`
+        // but there'll be a fair bit of plumbing needed to make each mode work
+
+        Ok(AzureBlobConfig {
+            metrics,
+            client,
+            cfg,
+            prefix,
+        })
+    }
+
+    /// Returns a new [AzureBlobConfig] for use in unit tests.
+    pub fn new_for_test() -> Result<Option<Self>, Error> {
+        let container_name = match std::env::var(Self::EXTERNAL_TESTS_AZURE_CONTAINER) {
+            Ok(container) => container,
+            Err(_) => {
+                if mz_ore::env::is_var_truthy("CI") {
+                    panic!("CI is supposed to run this test but something has gone wrong!");
+                }
+                return Ok(None);
+            }
+        };
+
+        let prefix = Uuid::new_v4().to_string();
+        let metrics = S3BlobMetrics::new(&MetricsRegistry::new());
+
+        let config = AzureBlobConfig::new(
+            EMULATOR_ACCOUNT.to_string(),
+            container_name.clone(),
+            prefix,
+            metrics,
+            Url::parse(&format!("http://localhost:40111/{}", container_name)).expect("valid url"),
+            Arc::new(ConfigSet::default()),
+        )?;
+
+        Ok(Some(config))
+    }
+
+    /// Returns a clone of Self with a new v4 uuid prefix.
+    pub fn clone_with_new_uuid_prefix(&self) -> Self {
+        let mut ret = self.clone();
+        ret.prefix = Uuid::new_v4().to_string();
+        ret
+    }
+}
+
+/// Implementation of [Blob] backed by Azure Blob Storage.
+#[derive(Debug)]
+pub struct AzureBlob {
+    metrics: S3BlobMetrics,
+    client: ContainerClient,
+    prefix: String,
+    _cfg: Arc<ConfigSet>,
+}
+
+impl AzureBlob {
+    /// Opens the given location for non-exclusive read-write access.
+    pub async fn open(config: AzureBlobConfig) -> Result<Self, ExternalError> {
+        if config.client.service_client().account() == EMULATOR_ACCOUNT {
+            // TODO: we could move this logic into the test harness.
+            // it's currently here because it's surprisingly annoying to
+            // create the container out-of-band
+            if let Err(e) = config.client.create().await {
+                warn!("Failed to create container: {e}");
+            }
+        }
+
+        let ret = AzureBlob {
+            metrics: config.metrics,
+            client: config.client,
+            prefix: config.prefix,
+            _cfg: config.cfg,
+        };
+
+        Ok(ret)
+    }
+
+    fn get_path(&self, key: &str) -> String {
+        format!("{}/{}", self.prefix, key)
+    }
+}
+
+#[async_trait]
+impl Blob for AzureBlob {
+    async fn get(&self, key: &str) -> Result<Option<SegmentedBytes>, ExternalError> {
+        let path = self.get_path(key);
+        let blob = self.client.blob_client(path);
+        let mut segments: Vec<MaybeLgBytes> = vec![];
+
+        let mut stream = blob.get().into_stream();
```

Comment:
According to the [docs `get`](https://docs.rs/azure_storage_blobs/0.21.0/azure_storage_blobs/prelude/struct.BlobClient.html#method.get) defaults to chunk sizes of 1MB. Mind adding a TODO to make this configurable or try tuning it?

Reply:
added a TODO